controllers/articles.js

```diff
@@ -153,6 +154,127 @@ export default class ArticleController {
     }
   }
 
+  /**
+   * @desc create a reaction
+   * @param {Object} articleSlug, the response obje
+   * @param {Object} userId - the response object
+   * @param {Object} type - the response object
+   * @memberof ArticleController
+   * @return {Object} returns an object
+   */
+  static async createReaction(articleSlug, userId, type) {
+    let createdReaction;
+    const likeArticle = await Reaction.findOne({
+      where: {
+        articleSlug,
+        userId,
+      },
+    });
+    if (!likeArticle) {
+      await Reaction.create({
+        articleSlug,
+        userId,
+        likes: type === 'like' ? true : false,
+      });
+      createdReaction = true;
+    } else {
+      await Reaction.destroy({
+        where: {
+          articleSlug,
+          userId,
+        }
+      });
+      createdReaction = false;
+    }
+    const allLikes = await Reaction.findAndCountAll({
+      where: {
+        articleSlug,
+        userId,
+        likes: type === 'like' ? true : false,
+      },
+    });
+    return { ...allLikes, createdReaction };
+  }
+
+  /**
+   * @desc check if a user likes an article
+   * @param {Object} req - the request object
+   * @param {Object} res - the response object
+   * @memberof ArticleController
+   * @return {Object} returns an object
+   */
+  static async likeArticle(req, res) {
+    try {
+      const { id: userId } = req.user;
+      const { slug: articleSlug } = req.params;
+      const getArticles = await Article.findOne({
```

Comment:
@Dubby20 what happens if the article doesn't exist?

Reply:
I have a middleware that checks if the article exists. If it doesn't, it returns 'Article not found'